### ISSUE TITLE
Postgres UUID columns not working?

### ISSUE THREAD

**Issue by acidjunk** [jwag956/flask-security#149]
Hi,

Thank for continuing development of Flask-Security: appreciated :-)

I tried to use Flask-Security-Too as a 'drop in' replacement. Forms, templates etc. work, but I think I am running into a problem with postgres UUID fields.

User model:
```
from sqlalchemy.dialects.postgresql import UUID

class User(db.Model, UserMixin):
    __tablename__ = 'user'
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True)
    email = Column(String(255), unique=True)
    first_name = Column(String(255), index=True)
    last_name = Column(String(255), index=True)
    username = Column(String(255), unique=True)
    password = Column(String(255))
    active = Column(Boolean())
    created_at = Column(DateTime, default=datetime.datetime.utcnow)
    confirmed_at = Column(DateTime())
    roles = relationship('Role', secondary='roles_users', backref=backref('users', lazy='dynamic'))
```
Console:
`sqlalchemy.exc.DataError: (psycopg2.DataError) invalid input syntax for uuid: "user@example.com"`

Looks as it doesn't see the UUID pk field correctly?

Configuration: 
https://github.com/acidjunk/improviser/blob/flask-security-too/improviser/main.py

Travis CI:
https://travis-ci.com/acidjunk/improviser/jobs/222120290

Or do I miss some new configuration stuff, somewhere?

**Comment by jwag956** [comment]
Thanks for the report - this is an area I spent a bunch of time on - there were many old issues around querying for different keys - along with the fact that psycopg2 is very picky about type mismatches. So I know just where to start looking :-)

**Comment by acidjunk** [comment]
Hi Chris, 

I can confirm that the login etc. now works as advertised when I pull it from master branch with commit SHA: 560a401

Small/Offtopic but, not sure if this is intended: Commit 33e7c4f breaks it, at least for my app:
`from flask_security.decorators import _get_unauthorized_response`

I will probably fix this on my side as it seems a bit hacky to rely on _get_unauthorized_response

Thanks for the quick fix :)

**Comment by jwag956** [comment]
Glad that is fixed.
That fix is available now on pypi as 3.3.0rc2

As for breaking - hmm - I usually consider _xxx as private - I changed the name as part of overhauling handling of authentication/authorization failures - which were sort of mushed together (please read the CHANGES since there is a backward incompatability change).

Note also that I have now made sure that ALL public/documented methods etc are exposed from the flask_security module - so you shouldn't have to import ANYTHING from flask_security.xyz. If you find some things you are using that aren't exposed or have '_' that you think SHOULD be useful public APIs - please let me know.
